tiledb/sm/filesystem/win.cc

```diff
@@ -62,8 +63,9 @@ namespace sm {
 
 namespace {
 /** Returns the last Windows error message string. */
-std::string get_last_error_msg() {
-  DWORD err = GetLastError();
+std::string get_last_error_msg(
+    decltype(GetLastError()) gle, const char* func_desc = "") {
+  DWORD err = gle;
   LPVOID lpMsgBuf;
   if (FormatMessage(
```

Comment:
I realize this is a comment on old code, but it would be useful to package up the `FormatMessage` call into a function that returns `std::string`. This and `GetLastError` are the only Windows system calls necessary here.

This SO question has numerous relevant details. Putting in a comment will allow much easier auditing of this code later.
https://stackoverflow.com/questions/455434/how-should-i-use-formatmessage-properly-in-c

Reply:
Not sure I follow, do you mean "package" for more general use elsewhere in tiledb?

Reply:
Not for general use, necessarily. Just a function for the present file that isolates the system call as narrowly as possible.

Reply:
It seems fairly narrowly isolated to me at the moment...
The original did have only one function, and that was all it did, 
but I split it so I had the possibility of passing in a gle value from some routines, where gle needed to be obtained before another API might destroy the one of interest...
Splitting the FormatMessage() part out here locally would only seem to me to add another, not useful, layer...?

Reply:
The issue is with exception safety. As a rule, Windows system calls should be isolated as narrowly as possible so that freeing system-allocated resources _always_ happens in immediate proximity to the call itself. As written currently, if there's an exception after the call to `FormatMessage` but before the end of the `get_last_error_msg`, `lpMsgBuf` will leak.